### PR TITLE
CQ-4346666 Ensure schema compliance of sling models as per AF 2.0 spe…

### PR DESCRIPTION
**Pull request by rismehta**
…cification

@review @vdua
DOD(yes)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java [hunk 1]
```diff
@@ -64,6 +64,11 @@ public String getThankYouMessage() {
         return thankyouMessage;
     }
 
+    @Override
+    public String getAdaptiveFormVersion() {
```
Comment (nit23uec) [inline bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java:68]: this should be picked up from storage (jcr here).

File: bundles/core/src/test/resources/schema/adaptive-form-properties.schema.json [hunk 2]
```diff
@@ -0,0 +1,322 @@
+{
+  "$id": "classpath:/schema/adaptive-form-properties.schema.json",
```
Comment (nit23uec) [inline bundles/core/src/test/resources/schema/adaptive-form-properties.schema.json:2]: this approach is not scalable. I'm assuming that we will fetch the publicly published schema in future.

**Review by nit23uec — CHANGES_REQUESTED** [review]
I see this flow like:
1. AF gets created with a particular schema version (which gets persisted on guide container).
this schema version is the same for which runtime (af-core) is there in AEM build.
2. Now the json generated from the sling models on that AEM build MUST comply to that particular schema version.
3. I can imagine a need of supporting multiple schema versions in the code (given the fact that we will have to support AFs created from all schema versions)

Do we intent to do the above task as part this jira or is there a separate action item for the same?

File: bundles/core/src/test/resources/schema/adaptive-form-properties.schema.json (hunk 2 above)
Comment (rismehta) [reply bundles/core/src/test/resources/schema/adaptive-form-properties.schema.json:2]: Yes, this is true for even the VS code plugin. Till the schema is public, we maintain a copy.

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java (hunk 1 above)
Comment (rismehta) [reply bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java:68]: No. Any changes in spec would mean a new version of form container. Hence the value needs to be hard coded

**Comment by rismehta** [comment]
> Now the json generated from the sling models on that AEM build MUST comply to that particular schema version

Right now, we don't validate the json at runtime(since we don't have to support to validate v7 json schema in  AEM yet

> I can imagine a need of supporting multiple schema versions in the code (given the fact that we will have to support AFs created from all schema versions)

I have changed the test case to incorporate schema version.

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java (hunk 1 above)
Comment (nit23uec) [reply bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java:68]: spec follows semantic versioning. the changes in minor/patch versions should not need another version of core component.

Comment (rismehta) [reply bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java:68]: That depends on the use-case. if we need to a new component, we would add it, else we could add the changes in the same component.

**Review by barshat7 — APPROVED** [review]

**Pull request merged**